packages/ui/src/machines/authenticator/index.ts

```diff
@@ -25,47 +25,58 @@ export function createAuthenticatorMachine() {
       context: {
         user: undefined,
         config: {},
-        services: {},
+        services: { ...defaultServices },
         actorRef: undefined,
       },
       states: {
         // See: https://xstate.js.org/docs/guides/communication.html#invoking-promises
         idle: {
-          on: {
-            INIT: {
+          invoke: {
+            src: 'getCurrentUser',
+            onDone: {
+              actions: 'setUser',
+              target: 'authenticated',
+            },
+            onError: {
               target: 'setup',
-              actions: 'configure',
             },
           },
         },
         setup: {
-          invoke: [
-            {
-              // TODO Wait for Auth to be configured
-              src: 'getCurrentUser',
-              onDone: {
-                actions: 'setUser',
-                target: 'authenticated',
+          initial: 'waitConfig',
+          states: {
+            waitConfig: {
+              on: {
+                INIT: {
+                  actions: 'configure',
+                  target: 'applyConfig',
+                },
               },
-              onError: [
+            },
+            applyConfig: {
```

Comment:
Once we have all the machine props ready, we apply those configs to get `loginMechanisms`, etc.

Reply:
👍